src/components/Button/Button.tsx

```diff
@@ -155,87 +153,78 @@ export default class Button extends React.Component<ButtonProps> {
+      iconPosition = ButtonIconPosition.LEFT,
+      status = ButtonStatus.ENABLED,
+    } = this.props;
 
     const href = (this.props as LinkButtonProps).href;
-    const status = (this.props as RegularButtonProps).status;
-    const isDisabled = status === ButtonStatus.DISABLED || status === ButtonStatus.LOADING;
+
+    const classNames = getClassNames({ status, iconPosition });
+
+    const buttonIcon = Icon && (
+      <span className={classNames.iconWrapper}>
+        <Icon />
+      </span>
+    );
+    const leftIcon = iconPosition === 'left' && buttonIcon;
+    const rightIcon = iconPosition === 'right' && buttonIcon;
 
     return (
       <BaseButton
         ariaLabel={ariaLabel}
-        className={`y-button ${className ? className : ''}`}
-        disabled={isDisabled}
+        className={`y-button ${className}`}
+        disabled={status !== ButtonStatus.ENABLED}
         href={href}
         type={href ? '' : type}
         onBlur={onBlur}
         onClick={onClick}
         onFocus={onFocus}
         onMouseEnter={onMouseEnter}
         onMouseLeave={onMouseLeave}
-        styles={getStyles(this.props)}
+        styles={getBaseButtonStyles({ size, color, fullWidth, status })}
       >
-        {this.getContents()}
-        {this.isLoading() && this.getSpinner()}
-      </BaseButton>
-    );
-  }
-
-  private getContents() {
-    const { icon: Icon, iconPosition, size, text, secondaryText } = this.props;
-
-    const textSize = size === ButtonSize.SMALL ? TextSize.SMALL : TextSize.MEDIUM_SUB;
-
-    const buttonIcon = Icon && (
-      <span className={`y-button--icon-wrapper-${iconPosition}`}>
-        <Icon />
-      </span>
-    );
-
-    const className = this.isLoading() ? 'y-button--content__hidden' : '';
-
-    const leftIcon = iconPosition === 'left' && buttonIcon;
-    const rightIcon = iconPosition === 'right' && buttonIcon;
-
-    return (
-      <Block textSize={textSize} className={className} textAlign="center">
-        {leftIcon}
-        {text}
-        {rightIcon}
-        <div className="y-button--secondary-text">
+        <Block
+          textSize={size === ButtonSize.SMALL ? TextSize.SMALL : TextSize.MEDIUM_SUB}
+          className={classNames.content}
+          textAlign="center"
+        >
           {leftIcon}
-          {secondaryText}
+          {text}
           {rightIcon}
-        </div>
-      </Block>
-    );
-  }
-
-  private getSpinner() {
-    const { color, size } = this.props;
-
-    const loadingText = (this.props as LoadingButtonProps).loadingText;
-
-    const spinnerColor = color === ButtonColor.PRIMARY ? SpinnerColor.DARK : SpinnerColor.LIGHT;
-    const spinnerSize = size === ButtonSize.SMALL ? SpinnerSize.XSMALL : SpinnerSize.SMALL;
-
-    return (
-      <span className="y-button--spinner">
-        <Spinner color={spinnerColor} size={spinnerSize} text={loadingText} hideText={true} isCentered={true} />
-      </span>
+          <div className={classNames.secondaryText}>
```

Comment:
I'm pretty sure we can make this block conditional on the presence of `secondaryText`, but am holding off since I didn't want to make this too hard to review.